hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/InvocationRegistry.java

```diff
@@ -289,25 +295,40 @@ private void scanHandleOperationTimeout() {
                 return;
             }
 
+            // todo: these 2 measurements should be added to the black-box.
+            int backupTimeouts = 0;
+            int invocationTimeouts = 0;
             for (Invocation invocation : invocations.values()) {
                 if (shutdown) {
                     return;
                 }
 
                 try {
-                    invocation.notifyInvocationTimeout();
+                    if (invocation.checkInvocationTimeout()) {
+                        invocationTimeouts++;
+                    }
                 } catch (Throwable t) {
                     inspectOutputMemoryError(t);
                     logger.severe("Failed to handle operation timeout of invocation:" + invocation, t);
                 }
 
                 try {
-                    invocation.checkBackupTimeout(backupTimeoutMillis);
+                    if (invocation.checkBackupTimeout(backupTimeoutMillis)) {
+                        backupTimeouts++;
+                    }
                 } catch (Throwable t) {
                     inspectOutputMemoryError(t);
                     logger.severe("Failed to handle backup timeout of invocation:" + invocation, t);
                 }
             }
+
+            log(backupTimeouts, invocationTimeouts);
+        }
+
+        private void log(int backupTimeouts, int invocationTimeouts) {
```

Comment:
Yes. And only when it has something sensible to say.. so if nothing got timeout --> nologging